iceberg-rust/src/arrow/write.rs

```diff
@@ -221,15 +227,21 @@ async fn create_arrow_writer(
         .map_err(|err| ArrowError::ExternalError(Box::new(err)))
         .unwrap();
 
-    let parquet_path =
-        partition_location.to_string() + &Uuid::now_v1(&rand).to_string() + ".parquet";
+    let prefix = rand[0..3]
+        .iter()
+        .fold(String::with_capacity(8), |mut acc, x| {
+            write!(&mut acc, "{:02x}", x).unwrap();
+            acc
+        });
 
-    let writer = BufWriter::new(object_store.clone(), parquet_path.clone().into());
+    let parquet_path = strip_prefix(data_location)
+        + &prefix
+        + "/"
+        + partition_path
```

Comment:
1) I would rather make this optional, most workloads won't benefit from this and you would rather have a nice layout in case you need to inspect your data manually
2) if we use the random prefix, we can get rid of the hive style partition path layout (`/a=10/`), it won't be usable anyway and will just make the key extra long
3) I think `write!(&mut acc, "{:x}", x).unwrap();` should be enough

Reply:
Thanks for the input, these are great points. I guess making it optional is the best way to go.

I will quickly sketch my reasoning. I would like this library to have good defaults, that users don't have to concern themselves with the partitioning of object storage. And it seems to me that parallelizing the requests with multiple S3 partitions per Iceberg partition has only benefits apart from the ugly file layout. Granted, this might not apply to all use cases but to some it will.

You are right that using a random prefix and the hive style layout probably doesn't make sense. I thought that maybe it would be a tradeoff between the two. But I guess you loose the structure once you introduce the random prefix and there is no point in keeping the partition values.

I will make this configurable with a table property, just like in Java.